### PR TITLE
[NUXE-640] Restructure Button kit stylings

### PR DESCRIPTION
**Pull request by bh247484**
#### What does this PR do?

1. Closes [this issue](https://github.com/powerhome/playbook/issues/1464) and resolves conflicting full width and dark mode styles.
2. Restructure button sass to be more modular.

#### Screens

Before:
before

After:
after

#### Breaking Changes

No prop changes

#### Runway Ticket URL

[NUXE-640](https://nitro.powerhrg.com/runway/backlog_items/NUXE-640)

#### How to test this

Will need alpha testing in Nitro.  Looks good locally but significant changes to Sass structure.

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **URGENCY** Please select a release milestone
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

**Comment by jasoncypret** [comment]
Why does it need Alpha testing? @bh247484

File: playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss [hunk 1]
```diff
@@ -68,7 +73,7 @@ $pb_button_border_width: 0px;
 
 // Primary ======================
 @mixin pb_button_primary {
-  @include pb_button;
+  @include variant_overrides;
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss:76]: It's kind of strange that `variant_overrides` doesn't pass any overrides. Thoughts? I feel a little less confident in my understanding when I see these. Should we change the name?

**Comment by bh247484** [comment]
> Why does it need Alpha testing? @bh247484

@jasoncypret First instinct was that the Sass changes were significant enough to justify alpha testing but actually if all the doc examples look good it's probably fine in Nitro.

No functionality/prop changes and styling in Nitro isn't significantly different than styling in Playbook docs

File: playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss [hunk 2]
```diff
@@ -12,24 +12,30 @@ $pb_button_h_padding: 34px;
 $pb_button_hover_darken: 4%;
 $pb_button_border_width: 0px;
 
+@mixin variant_overrides($bg: $primary_action, $color: $text_dk_default, $border: transparent) {
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss:15]: Could we maybe call this mixin `pb_button_variant` so it's clear throughout the file?

```suggestion
@mixin pb_button_variant($bg: $primary_action, $color: $text_dk_default, $border: transparent) {
```

File: playbook/dist/reset.css [hunk 3]
```diff
@@ -1 +1,61 @@
-*{box-sizing:border-box;margin:0;padding:0}*:before,*:after{box-sizing:border-box}html{-webkit-tap-highlight-color:rgba(0,0,0,0);height:100vh;overflow-x:hidden}body{font-family:"Proxima Nova","Helvetica Neue",Helvetica,Arial,sans_serif;font-size:16px;line-height:1.5;background-color:#F3F7FB;height:100%;letter-spacing:0;font-weight:400;font-style:normal;text-rendering:optimizeLegibility;-moz-font-feature-settings:"liga" on;color:#242B42;margin:0 !important;padding:0 !important;box-sizing:border-box;min-height:100vh;padding:50px}a{text-decoration:none;color:#0056CF}
+/* CLEAN UP AND REMOVE */
```
Comment (jasoncypret) [inline playbook/dist/reset.css:1]: I guess this gets cleaned up with a release?

File: playbook/dist/reset.css [hunk 4]
```diff
@@ -1 +1,61 @@
...
+  margin: 0 !important;
+  padding: 0 !important;
+  box-sizing: border-box;
+  min-height: 100vh;
+  padding: 50px; }
+
+a {
+  text-decoration: none;
+  color: #0056CF; }
+
```
Comment (thestephenmarshall) [inline playbook/dist/reset.css:61]: 🤔 How did this get updated to be so pretty? 🌹

**Review by jasoncypret — CHANGES_REQUESTED** [review]
I'm not sure why yarn lock and other files are included here. CC @xjunior

**Review by thestephenmarshall — APPROVED** [review]
None of my comments are super important here. Looks good and I will side with whatever you think regarding alpha testing.

File: playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss (hunk 2 above)
Comment (bh247484) [reply playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss:15]: @thestephenmarshall Good call.  Struggled to come up with an intuitive name for this one.

I'll switch it to `pb_button_variant`.

File: playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss (hunk 1 above)
Comment (bh247484) [reply playbook/app/pb_kits/playbook/pb_button/_button_mixins.scss:76]: Yeah it's odd, the mixin itself has default arguments.  Unfortunately need them for some mixin implementations where arguments are specified.

I will see if I can just pass them explicitly on line 76 above though so it's more intuitive.

File: playbook/dist/reset.css (hunk 3 above)
Comment (bh247484) [reply playbook/dist/reset.css:1]: Yeah this is a strange one.  Meant to bring it up at dailies.

The new repo setup re-compiles the dist when you make changes.  It also re-compiles the `reset.css`, but probably doesn't need to.  Sometimes webpack refused to compile because of some problem with `reset.css`, then I'd have to restart the server.  Didn't have the time to search through webpack config to track down what's going on.

TLDR: we may need to take a look at what's getting compiled when.

File: playbook/dist/reset.css (hunk 4 above)
Comment (bh247484) [reply playbook/dist/reset.css:61]: Yeah this is a strange one.  Meant to bring it up at dailies.

The new repo setup re-compiles the dist when you make changes.  It also re-compiles the `reset.css`, but probably doesn't need to.  Sometimes webpack refused to compile because of some problem with `reset.css`, then I'd have to restart the server.  Didn't have the time to search through webpack config to track down what's going on.

TLDR: we may need to take a look at what's getting compiled when.

Comment (thestephenmarshall) [reply playbook/dist/reset.css:61]: From what I'm gathering (seeing other PRs) this is normal, so I would let it go and if it goes splat then Carlos can take a look.
